lib/bolt/plugin.rb

```diff
@@ -27,19 +41,32 @@ def self.setup(config, pdb_client)
       plugins
     end
 
-    def initialize(_config)
+    def initialize(config, analytics)
+      @config = config
+      @analytics = analytics
       @plugins = {}
+      @reported = Set.new
     end
 
     def add_plugin(plugin)
       @plugins[plugin.name] = plugin
     end
 
-    def for_hook(hook)
-      @plugins.filter { |_n, plug| plug.hooks.include? hook }
+    def get_hook(plugin_name, hook)
+      plugin = by_name(plugin_name)
+      raise PluginError::Unknown, plugin_name unless plugin
+      raise PluginError::UnsupportedHook.new(plugin_name, hook) unless plugin.respond_to?(hook)
+      @analytics.report_bundled_content("Plugin #{hook}", plugin_name)
+
+      plugin.method(hook)
     end
 
     def by_name(plugin_name)
+      unless @reported.include?(plugin_name)
+        @reported << plugin_name
+        @analytics.report_bundled_content('Plugin', plugin_name)
```

Comment:
Why do we submit this twice? The version with the hook seems like it has a superset of this information. Will this show up as 2 plugin uses in analytics?

Reply:
this was a bug left over from the initial implementation.